tfe/resource_tfe_team_access.go

```diff
@@ -166,3 +345,33 @@ func resourceTFETeamAccessImporter(d *schema.ResourceData, meta interface{}) ([]
 
 	return []*schema.ResourceData{d}, nil
 }
+
+func setCustomOrComputedPermissions(d *schema.ResourceDiff, meta interface{}) error {
+	if _, ok := d.GetOk("access"); ok {
+		if d.HasChange("access") {
+			_, n := d.GetChange("access")
+			new := n.(string)
+
+			if new != "custom" {
```

Comment:
⚠️ Martin brought up that this might be problematic for when some sort of interpolation is done that cannot be known at plan-time. I don't really think there's any way around this limitation, and at this point think it's reasonable to expect that changes to resources that affect authorization in the application probably should _not_ be done only at apply-time anyway ;)

Reply:
It is in theory possible to address it by calling [`d.NewValueKnown("custom")`](https://godoc.org/github.com/hashicorp/terraform/helper/schema#ResourceDiff.NewValueKnown) and doing something special if it returns `false`, but I'm not really sure _what_ you'd do if it were false... that would indicate that at apply time the value could be either `custom` or not `custom`, which would mean that `permissions` could either be "computed" (unknown) or not.

One way we could avoid potential into-corner-painting :paintbrush: here would be to make an explicit block of this argument being unknown, so users who do this can see explicitly that it isn't allowed. That is, to _begin_ this function with an early exit for that case:

```go
if !d.NewValueKnown("custom") {
    return fmt.Errorf("'access' cannot be derived from a value that is unknown during planning")
}
```

This error message is not particularly actionable, but I think that can be justified by the fact that this situation ought to be very rare and so this test is there only as a safety to prevent something odd/unclear happening in that unlikely event, and prevent that odd behavior from potentially becoming a compatibility constraint.

Reply:
can you explain what this means a little more? do you have an example of interpolation that can't be known at plan time? i'm sorry, i'm just having a hard time wrapping my head around this and what sort of things we should be looking out for.

Reply:
@lafentres With data resources that have computed values, the read is deferred until the apply phase - so if you were to do an interpolation involving that to determine the access level here, that would be an example of something where the logic here (which needs to be `"custom"` at plan time) would break.

I like Martin's suggestion of avoiding it altogether by restricting that behavior with an error, good idea!